benchkit/app.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from multiprocessing import Semaphore
+
+import typing_extensions as te
+from sanic import Sanic
+from sanic.config import Config
+from sanic.exceptions import (
+    BadRequest,
+    NotFound,
+)
+from sanic.request import Request
+from sanic.response import (
+    empty,
+    HTTPResponse,
+    text,
+)
+
+from .context import BenchKitContext
+from .env import env
+from .workloads import Workload
+
+
+T_App: te.TypeAlias = "Sanic[Config, BenchKitContext]"
+
+
+def create_app() -> T_App:
+    app: T_App = Sanic("Python_BenchKit", ctx=BenchKitContext())
+
+    @app.main_process_start
+    async def main_process_start(app: T_App) -> None:
+        app.shared_ctx.running = Semaphore(1)
```

Comment:
I'm curious about the function of this semaphore.

Reply:
This is to make sure that only one worker process is started. The backend shouldn't use multiple drivers and each process would get its own driver instance. That's basically just a sanity check that the server is started with `workers=1` and that that option really does what it claims to do.